src/utils/CustomPropTypes.js

```diff
@@ -57,4 +66,18 @@ function createMountableChecker() {
   return createChainableTypeChecker(validate);
 }
 
+function createKeyOfChecker(obj) {
+  function validate(props, propName, componentName) {
+    let propValue = props[propName];
+    if (!obj.hasOwnProperty(propValue)) {
+      let valuesString = JSON.stringify(Object.keys(obj));
+      return new Error(
+        ('Invalid prop `' + propName + '` of value `' + propValue + '` ') +
+        ('supplied to `' + componentName + '`, expected one of ' + valuesString + '.')
```

Comment:
Would that be preferable even with needing to escape the backticks within the string? I basically copied the code from `ReactPropTypes` `createOneOfChecker`. The surrounding code also doesn't use interpolation.